fast_llm/data/preparator/gpt_memmap/prepare.py

```diff
@@ -321,6 +324,12 @@ def run(self) -> None:
 
         # Calculate total number of tokens
         total_tokens = sum(tqdm.tqdm(tokenized_dataset["num_tokens"], desc="Counting tokens", unit="tokens"))
+        total_pixels = (
+            sum(tqdm.tqdm(tokenized_dataset["num_pixels"], desc="Counting pixels", unit="pixels"))
+            if self._config.dataset.images
+            else 0
+        )
+        total_tokens += total_pixels // np.dtype(self._data_type.numpy).itemsize
```

Comment:
Should divide by `(image_patch_size**2)` instead?

Reply:
This is something I couldn't cleanly solve. `total_tokens` here is used to determine the shard size, so I got the token-equivalent bytes for pixels. Everywhere else we use `image_patch_size**2` to determine the actual number of image tokens and weigh the shard accordingly for blending

Reply:
Oh I see, that makes sense yes! Added a comment as a suggestion
```suggestion
        # Add the token-equivalent bytes of pixels to determine shard size
        total_tokens += total_pixels // np.dtype(self._data_type.numpy).itemsize
```